src/schema/users.ts

```diff
@@ -2088,6 +2190,42 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
         coresRole: userCoresRole,
       };
     },
+    userJobPreferences: async (
+      _,
+      __,
+      ctx: AuthContext,
+    ): Promise<GQLUserJobPreferences> => {
+      // important: if you need to implement public user job preferences queries, we need to remove the "currentTotalComp" for other users
```

Comment:
Usually we would do this in graphorm, with computed field where we check if user requesting is the same as user owning job.